include/clang/Lex/MacroInfo.h

```diff
@@ -515,6 +510,9 @@ class ModuleMacro : public llvm::FoldingSetNode {
     ID.AddPointer(II);
   }
 
+  /// Get the name of the macro.
+  IdentifierInfo *getName() const { return II; }
```

Comment:
@zygoloid, I'll bring this to Clang proper if it makes sense, but *does* it make sense? It seems like if we have this information it should be reasonable to ask for it, but if not I'll figure out another way to refer to "this particular macro definition by this name" in our lookup table. (Which doesn't *have* to keep pointer-sized entries, but it'd be nice if it could.)